Backend354/src/main/java/com/project/backend354/controller/ImageController.java

```diff
@@ -39,4 +44,34 @@ public ResponseEntity<ImageUploadResponse> uploadImage(@RequestParam MultipartFi
 
         return ResponseEntity.ok(response);
     }
-}
+
+    @GetMapping("/download/{filename}")
+    public ResponseEntity<Resource> downloadImage(@PathVariable String filename) {
+        Resource resource = downloadService.getFileResource(filename);
+
+        if (resource == null) {
```

Comment:
Don't use null as a way to manage process flow, just throw an exception in the service if you want to make an operation fail